fastcrypto-cli/src/ecvrf.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use clap::Parser;
+use fastcrypto::vrf::ecvrf::{ECVRFKeyPair, ECVRFPrivateKey, ECVRFProof, ECVRFPublicKey};
+use fastcrypto::vrf::{VRFKeyPair, VRFProof};
+use rand::thread_rng;
+use std::io::{Error, ErrorKind};
+
+#[derive(Parser)]
+#[command(name = "ecvrf-cli")]
+#[command(about = "Elliptic Curve Verifiable Random Function (ECVRF) over Ristretto255 according to draft-irtf-cfrg-vrf-15.", long_about = None)]
+enum Command {
+    /// Generate a key pair for proving and verification.
+    Keygen,
+
+    /// Create an output/hash and a proof.
+    Prove(ProveArguments),
+
+    /// Verify an output/hash and a proof.
+    Verify(VerifyArguments),
+}
+
+#[derive(Parser, Clone)]
+struct ProveArguments {
+    /// The hex encoded input string. May be of arbitrary length but should ideally be at least 16 bytes.
+    #[clap(short, long)]
+    input: String,
+
+    /// A hex encoding of the secret key. Corresponds to a scalar in Ristretto255 and must be 32 bytes.
+    #[clap(short, long)]
+    secret_key: String,
+}
+
+#[derive(Parser, Clone)]
+struct VerifyArguments {
+    /// Hex-encoded Sha512 hash of the proof. Must be 64 bytes.
+    #[clap(short, long)]
+    output: String,
+
+    /// Encoding of the proof to verify. Must be 80 bytes.
+    #[clap(short, long)]
+    proof: String,
+
+    /// Hex encoding of the input string used to generate the proof. May be of arbitrary length but
+    /// should ideally be at least 16 bytes.
+    #[clap(short, long)]
+    input: String,
+
+    /// The public key corresponding to the secret key used to generate the proof.
+    #[clap(short, long)]
+    verification_key: String,
+}
+
+fn main() {
+    match execute(Command::parse()) {
+        Ok(res) => {
+            println!("{}", res);
+            std::process::exit(exitcode::OK);
+        }
+        Err(e) => {
+            println!("Error: {}", e);
+            std::process::exit(exitcode::DATAERR);
+        }
+    }
+}
+
+fn execute(cmd: Command) -> Result<String, std::io::Error> {
+    match cmd {
+        Command::Keygen => {
+            let keypair = ECVRFKeyPair::generate(&mut thread_rng());
+            let sk_string =
+                hex::encode(bincode::serialize(&keypair.sk).map_err(|_| {
+                    Error::new(ErrorKind::Other, "Failed to serialize secret key.")
+                })?);
+            let pk_string =
+                hex::encode(bincode::serialize(&keypair.pk).map_err(|_| {
+                    Error::new(ErrorKind::Other, "Failed to serialize public key.")
+                })?);
+
+            let mut result = "Secret key: ".to_string();
+            result.push_str(&sk_string);
+            result.push_str("\nPublic key: ");
```

Comment:
Verification Key (like the name used below)

Reply:
I've used public key everywhere instead to align with the fastcrypto api.